### PR TITLE
Demonstration for gimbal based camera

### PR DESCRIPTION
**Pull request by GilesBathgate**
### Description

Demonstrates the gimbal based camera idea I was proposing #347 

Try it out with:
`./MK404 Prusa_MK3 -g fancy`

P.S ignore the variable names, I was basing them on printer coordinate system (Z=up) not your viewer coordinate system (y=UP)

### CODE REVIEW

**Comment by vintagepc** [comment]
Sorry I haven't had a chance to look at this yet... lots of projects in the air at the moment :)

**Comment by GilesBathgate** [comment]
@vintagepc Yeah ok. I might do some more work on it anyway, I just didn't want to continue if it's not to your liking.

**Comment by vintagepc** [comment]
Is the loss of pan/zoom deliberate or just something that hasn't been implemented yet?

I did toy with it a bit and while it does feel a bit more natural in some aspects I'm not sure a complete replacement is ideal for folks already used to the existing camera, and it's probably better to leave a command line option to choose between both control sets.

File: utility/MK3SGL.h [hunk 1]
```diff
@@ -186,6 +187,12 @@ class MK3SGL: public BasePeripheral, public Scriptable, private IKeyClient
 			m_bSDCard = {true},
 			m_bPrintSurface = {true};
 
+		float xrotate=0.f;
```
Comment (vintagepc) [inline utility/MK3SGL.h:190]: Please follow the established variable naming conventions with the appropriate Hungarian notation prefixes.

**Comment by GilesBathgate** [comment]
> Is the loss of pan/zoom deliberate or just something that hasn't been implemented yet?

Just not implemented yet.

>leave a command line option to choose between both control sets.

Fair enough. I'll create another PR then.

**Pull request closed without merging**